tests/e2e/base-ia.test.js

```diff
@@ -0,0 +1,139 @@
+/* global fixture */
+import { ClientFunction, Selector } from 'testcafe';
+
+// Uses same plugins as archive.org
+fixture `Tests IA BookReader interactions`.page('http://localhost:8000/BookReaderDemo/demo-ia-plato.html');
+
+test('On load, pages fit fully inside of the BookReader™', async t => {
+  const brShell = await Selector('.BookReader');
+  const brContainer = await brShell('.BRcontainer');
+
+  await t.expect(brShell.visible).ok();
+  await t.expect(brContainer.visible).ok();
+
+  const shellHeight = await brShell.getBoundingClientRectProperty('height');
+  const bookHeight = await brContainer.getBoundingClientRectProperty('height');
+  await t.expect(bookHeight).lte(shellHeight, 'images do not get cropped vertically');
+
+  const shellWidth = await brShell.getBoundingClientRectProperty('width');
+  const bookWidth = await brContainer.getBoundingClientRectProperty('width');
+  await t.expect(bookWidth).lte(shellWidth, 'images do not get cropped horizontally');
+});
+
+test('Nav menu displays properly', async t => {  
+  await t.expect((await Selector('.BRfooter')).visible).ok();
+  await t.expect((await Selector('.BRicon.book_left')).visible).ok();
+  await t.expect((await Selector('.BRicon.book_right')).visible).ok();
+  await t.expect((await Selector('.BRicon.onepg')).visible).ok();
+  await t.expect((await Selector('.BRicon.twopg')).visible).ok();
+  await t.expect((await Selector('.BRicon.thumb')).visible).ok();
+  // Need .desktop-only class as blank zoom buttons in mobile menu
+  await t.expect((await Selector('.BRicon.zoom_in.desktop-only')).visible).ok();
+  await t.expect((await Selector('.BRicon.zoom_out.desktop-only')).visible).ok();
+  await t.expect((await Selector('.BRicon.full')).visible).ok();
+});
+
+// Need to disable page caching to have cookies persist in test
+test.disablePageCaching('Clicking `page flip buttons` updates location', async t => {
+  const hash = ClientFunction(() => document.location.hash);
+  const flipRight = await Selector('.book_right');
+  const flipLeft = await Selector('.book_left');
+  // No hash
+  await t.expect(hash()).eql('');
+  // Page navigation creates hash
+  await t.click(flipRight);
+  await t.expect(hash()).contains('#page/n');
+  await t.expect(hash()).contains('/mode/2up');
+  await t.click(flipLeft);
+  await t.expect(hash()).contains('#page/n');
+  await t.expect(hash()).contains('/mode/2up');
+  // Cookie set by page navigation creates hash on navigate to canonical URL
+  await t.navigateTo('demo-ia-plato.html');
+  await t.expect(hash()).contains('#page/n');
+  await t.expect(hash()).contains('/mode/2up');
```

Comment:
Can we test something else here when you click on any selector? e.g. click on 1up or thumbs view and then see if respected view mode is appearing on URL. Thanks!

Reply:
Thanks for the suggestion. All kinds of tests can be added. I copied the existing base test, fixing it for the mobile menu. I only added tests for this specific Jira/issue, code changes.

There will likely be more code changes and we should add more tests for that code.